### PR TITLE
Add ESLint and StyleLint

### PR DESCRIPTION
**Pull request by Saibamen**
To have autofix on save in Visual Studio Code, add this to your `settings.json` in `.vscode` folder:

```json
{
  "editor.codeActionsOnSave": {
    "source.organizeImports": true,
    "source.fixAll.eslint": true
  },
}
```

Plugin: https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint

### CODE REVIEW

**Comment by Saibamen** [comment]
@louislam: Which quotes do you prefer?
Single (`'`) or double (`"`)?

**Comment by louislam** [comment]
> @louislam: Which quotes do you prefer?
> Single (`'`) or double (`"`)?

I prefer not to strict the type of string quote.

I personally prefer double quotes as it is easier to switch to other languages such as Java/PHP for myself. However, a lot of JS example source code that copy from the Internet is single quote.

File: src/components/HeartbeatBar.vue [hunk 1]
```diff
@@ -91,14 +75,14 @@ export default {
 
             let width
             if (this.maxBeat > 0) {
-                width = (this.beatWidth + this.beatMargin * 2) * this.maxBeat + (leftRight * 2) + "px"
+                width = `${(this.beatWidth + this.beatMargin * 2) * this.maxBeat + (leftRight * 2)}px`
```
Comment (louislam) [inline src/components/HeartbeatBar.vue:78]: I also prefer not to strict to a string concatenation style. It is an example of bad use of ${}.  It is harder to read and harder to write than the "Plus" sign.

File: src/util-frontend.js [hunk 2]
```diff
@@ -396,20 +396,20 @@ export function timezoneList() {
...
+            console.log('Skip this timezone')
         }
 
     }
 
     result.sort((a, b) => {
         if (a.time > b.time) {
             return 1;
-        } else if (b.time > a.time) {
+        } if (b.time > a.time) {
```
Comment (louislam) [inline src/util-frontend.js:407]: Look strange to me.

File: src/layouts/Layout.vue [hunk 3]
```diff
@@ -1,87 +1,157 @@
...
+        >
+            <div>➕</div>Add
+        </router-link>
+        <router-link
+            to="/settings"
+            class="nav-link"
+            @click="$root.cancelActiveList"
+        >
+            <div>🔧</div>Settings
+        </router-link>
```
Comment (louislam) [inline src/layouts/Layout.vue:124]: Multiple lines of attribute is not good in this case.

Comment (chakflying) [reply src/layouts/Layout.vue:124]: While it does look a bit weird in this case, I do find it helpful when we inevitably add a bunch of classes and make the line super long. I don't know if there's an intermediate setting, perhaps allow more chars per line?

File: src/util-frontend.js (hunk 2 above)
Comment (Saibamen) [reply src/util-frontend.js:407]: Sorry, I forget to add new line

The case: Don't use `else` if we `return` in previous `IF` statement

File: src/layouts/Layout.vue (hunk 3 above)
Comment (Saibamen) [reply src/layouts/Layout.vue:124]: This is the recommended settings from `plugin:vue/vue3-recommended`

File: src/components/HeartbeatBar.vue (hunk 1 above)
Comment (Saibamen) [reply src/components/HeartbeatBar.vue:78]: I commented the `'prefer-template': 'error',` rule

**Comment by Saibamen** [comment]
> I personally prefer double quotes as it is easier to switch to other languages such as Java/PHP for myself. However, a lot of JS example source code that copy from the Internet is single quote.

That is why this rule has warning, not error. I changed it to `double`. Check my first post to see how to enable autofix on save in VS Code

File: src/layouts/Layout.vue (hunk 3 above)
Comment (Saibamen) [reply src/layouts/Layout.vue:124]: https://eslint.vuejs.org/rules/max-attributes-per-line.html

I can override default settings. Just give me numbers for https://eslint.vuejs.org/rules/max-attributes-per-line.html#options

Comment (louislam) [reply src/layouts/Layout.vue:124]: Prefer to off, it should be determined by human.

"vue/max-attributes-per-line": "off"

Comment (Saibamen) [reply src/layouts/Layout.vue:124]: >While it does look a bit weird in this case, I do find it helpful when we inevitably add a bunch of classes and make the line super long.

And git diff will be more readable

Comment (Saibamen) [reply src/layouts/Layout.vue:124]: > Prefer to off, it should be determined by human.
> 
> "vue/max-attributes-per-line": "off"

OK

**Comment by Saibamen** [comment]
Ready for Code Review

**Comment by louislam** [comment]
image

/server/database.js

```
Parsing error: Unexpected token =
```

Any ideas?

**Comment by Saibamen** [comment]
> /server/database.js
> 
> ```
> Parsing error: Unexpected token =
> ```
> 
> Any ideas?

No

**Comment by chakflying** [comment]
> 
> 
> image
> 
> /server/database.js
> 
> ```
> Parsing error: Unexpected token =
> ```
> 
> Any ideas?

Should be fixed in #121.

**Comment by Zaid-maker** [comment]
Hi

**Pull request merged**